dev-libs/libsecp256k1/libsecp256k1-0.2.0-r2.ebuild

```diff
@@ -0,0 +1,82 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit autotools multilib-minimal
+
+MyPN=secp256k1
+DESCRIPTION="Optimized C library for EC operations on curve secp256k1"
+HOMEPAGE="https://github.com/bitcoin-core/secp256k1"
+SRC_URI="
+	https://github.com/bitcoin-core/secp256k1/archive/v${PV}.tar.gz -> ${P}.tar.gz
+	https://github.com/bitcoin-core/secp256k1/commit/772e747bd9104d80fe531bed61f23f75342d7d63.patch -> ${PN}-PR1159-772e74.patch
+"
+
+LICENSE="MIT"
+SLOT="0/1"  # subslot is "$((_LIB_VERSION_CURRENT-_LIB_VERSION_AGE))" from configure.ac
+KEYWORDS="~amd64 ~arm ~arm64 ~mips ~ppc ~ppc64 ~x86 ~amd64-linux ~x86-linux"
+IUSE="+asm +ecdh experimental +extrakeys lowmem +recovery +schnorr static-libs test valgrind"
+RESTRICT="!test? ( test )"
+
+REQUIRED_USE="
+	asm? ( || ( amd64 arm ) arm? ( experimental ) )
+	schnorr? ( extrakeys )
+"
+
+BDEPEND="
+	sys-devel/autoconf-archive
+	virtual/pkgconfig
+	valgrind? ( dev-util/valgrind )
+"
+
+PATCHES=(
+	"${DISTDIR}/${PN}-PR1159-772e74.patch"
+	"${FILESDIR}/0.2.0-fix-out-of-source-tree-build.patch"
+)
+
+S="${WORKDIR}/${MyPN}-${PV}"
+
+src_unpack() {
+	unpack "${P}.tar.gz"
+}
+
+src_prepare() {
+	default
+	eautoreconf
+
+	# Generate during build
+	rm -f src/precomputed_ecmult.c src/precomputed_ecmult_gen.c || die
+}
+
+multilib_src_configure() {
+	local myconf=(
+		$(use_enable static{-libs,})
+		--disable-benchmark
+		$(use_enable experimental)
+		$(use_enable test tests)
+		$(use_enable test exhaustive-tests)
+		$(use_enable {,module-}ecdh)
+		$(use_enable {,module-}extrakeys)
+		$(use_enable {,module-}recovery)
+		$(use_enable schnorr module-schnorrsig)
+		$(usev lowmem '--with-ecmult-window=4 --with-ecmult-gen-precision=2')
+		$(use_with valgrind)
+	)
+	if use asm; then
+		if use arm; then
+			myconf+=( --with-asm=arm )
+		else
+			myconf+=( --with-asm=auto )
+		fi
+	else
+		myconf+=( --with-asm=no )
+	fi
+
+	ECONF_SOURCE="${S}" econf "${myconf[@]}"
```

Comment:
Not that it really matters, but this changes myconfargs -> myconf - and we lose some other changes from the ebuild in ::gentoo.

Can you go over the diff and make sure nothing else we want to keep is lost? (I deliberately didn't keep the experimental asm for arm, for example, as I felt it wasn't really worth it.)

Reply:
Oh, I see that part is included in the aesthetics commit. I'd prefer that some stuff like that is left as-is normally unless someone is going to take it over, bu then again I'm hoping you will, so that's fine.

But please check the arm bit, I'd rather just adjust that to what's in tree right now.

Reply:
The change in variable name was to reduce the diff versus the version in the `bitcoin-gentoo` repo and also to save a few bytes. I'll change it back. (Not worth bike shedding over.)

Can you be more specific about what was lost? I manually made changes to the ebuild in this repo, and I reviewed the diffs of all of my commits carefully. I don't believe anything was lost. (Some things were removed intentionally because they shouldn't have been there.)

Regarding experimental for ARM assembly, if you omit that from `REQUIRED_USE`, then users will get build-time errors in `configure`. It's better that `REQUIRED_USE` accurately mirrors the constraints imposed by `configure`.

Reply:
(I don't mind you making some style changes like this if you intend on making further PRs for Bitcoin stuff in ::gentoo. I don't really want to handle it myself as I don't use it daily or anything.)

Sorry, I see now, one still needs --enable-experimental to build with ARM asm. I guess it's fine to keep it as you've done then, even if I'd personally just drop it entirely until it's not experimental given the nature of the library.

Please disregard.

Reply:
I'll keep submitting PRs for Bitcoin-related stuff if you keep merging them. One of these days I should probably just go through the process to become a Gentoo dev. I've been doing this for almost two decades now. 😅

Reply:
I'm happy on both counts then! Happy to have a chat about the latter on IRC at some point (email is fine too, I just tend to prefer the conversational nature of IRC).